### ISSUE TITLE
UndefinedPropertyFetch in correctly resolved type

### ISSUE THREAD

**Issue by vudaltsov** [vimeo/psalm#7436]
https://psalm.dev/r/894d55f18b

**Comment by orklah** [comment]
This solves the issue: https://psalm.dev/r/96655a72f0

The actual issue in Psalm seems related to intersection types and it's badly handled:
https://github.com/vimeo/psalm/blob/b9e3979c3f285ed1c25b72b5490dea53f7107c30/src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/AtomicPropertyFetchAnalyzer.php#L1004

We enter in this method when we detected we have an intersection with an interface. The line 1004 just takes the first object in the intersection and rolls with it. Unfortunately, in your snippet it's A, and A doesn't have the property so it fails.

**Comment by vudaltsov** [comment]
Yes, I know that single check works. But the actual code needs both, because there are three code branches. I omitted the code for simplicity.
